### ISSUE TITLE
Shared parameters are dereferenced too late for use in code sample templates

### ISSUE THREAD

**Issue by MikeRalphson** [Mermade/widdershins#9]
As noticed by ~~@bc-alyssNoland~~ @PreciselyAlyss 
https://github.com/MikeRalphson/bigcommerce-api/blob/openapi/swagger/v3-api.yaml

**Comment by MikeRalphson** [comment]
@bc-AlyssNoland the fix passed the test suite, so can you try it out from the repository version, or would you need a new version published to npm? With your example Python code template you are probably better off using the `parameter.safeType` pseudo-property as this normalises a few quirks.

**Comment by bc-AlyssNoland** [comment]
@MikeRalphson The python template you made is working for create requests, but the parameters aren't making it to any other requests.

**Comment by MikeRalphson** [comment]
Hi @bc-AlyssNoland from that screenshot I can't quite see which post request is missing parameters. Could you let me have the /path ?

**Comment by bc-AlyssNoland** [comment]
https://gist.github.com/bc-AlyssNoland/2f64aa9041a794f93712d8b4a6a4649a

I posted the output of the file in the gist above. It seems some GET and DELETE requests are getting the parameters. I agree with only listing things that are required. We (my company) actually use an extension to indicate which fields are required by operation, but I can work on modifying a fork to accommodate that. 

```yaml
amount:
        type: number
        format: double
        description: |
          The value of the adjustment by the bulk pricing rule.
        x-required:
          - post
          - put
        minimum: 0
```

**Comment by MikeRalphson** [comment]
@bc-AlyssNoland Thanks, will take a look at the gist.

You should be able to use the new template callback feature to flip the `parameter.required` value based on your vendor extension if you want to avoid forking in this case.

**Comment by MikeRalphson** [comment]
@PreciselyAlyss I believe the underlying bug has been fixed and is in v1.1.0 - please feel free to re-open this issue if there is still something outstanding, or to raise a new one. Thanks.
